keep/api/core/db.py

```diff
@@ -4482,3 +4487,43 @@ def get_resource_ids_by_resource_type(
         # Execute the query and return results
         result = session.exec(query)
         return result.all()
+
+
+def get_last_alert_by_fingerprint(
+    tenant_id: str, fingerprint: str, session: Optional[Session] = None
+) -> Optional[LastAlert]:
+    with existed_or_new_session(session) as session:
+        return session.exec(
+            select(LastAlert)
+            .where(
+                and_(
+                    LastAlert.tenant_id == tenant_id,
+                    LastAlert.fingerprint == fingerprint,
+                )
+            )
+        ).first()
+
+def set_last_alert(
+    tenant_id: str, alert: Alert, session: Optional[Session] = None
+) -> None:
+    last_alert = get_last_alert_by_fingerprint(tenant_id, alert.fingerprint, session)
```

Comment:
in theory the `alert.fingerprint` could be None. Should we handle it? 
```suggestion
    if not alert.fingerprint:
        raise AlertEmptyFingerprint()
    last_alert = get_last_alert_by_fingerprint(tenant_id, alert.fingerprint, session)
```

Reply:
As far as I understand it was possible until end of September. Last entry in prod database is 2023-09-26 12:16:21.000